eth_utils/abi.py

```diff
@@ -448,20 +448,22 @@ def get_normalized_abi_inputs(
     if duplicate_args:
         raise TypeError(
             f"{abi_element.get('name')}() got multiple values for argument(s) "
-            f"'{', '.join(duplicate_args)}'"
+            f"'{', '.join(duplicate_args)}'."
         )
 
     # Check for unknown args
-    unknown_args = kwarg_names.difference(sorted_arg_names)
+    # Arg names sorted to raise consistent error messages
+    unknown_args = tuple(sorted(kwarg_names.difference(sorted_arg_names)))
```

Comment:
`unknown_args` was thrashing tests after I updated them to match the exception message. This keeps keys sorted after extracting them from a dict so the message always returns names in the same order.